src/parser/jobs/ast/modules/CelestialOpposition.tsx

```diff
@@ -47,40 +55,44 @@ export default class CelestialOpposition extends Module {
 		this.lastUse = event.timestamp
 	}
 
-	private onSect(event: BuffEvent) {
-		this.activeSect = getDataBy(this.data.statuses, 'id', event.ability.guid)
+	private onSect(event: Events['statusApply']) {
+		if (event.status === this.data.statuses.DIURNAL_OPPOSITION.id) {
+			this.activeSect = this.data.actions.DIURNAL_SECT.id
+		} else {
+			this.activeSect = this.data.actions.NOCTURNAL_SECT.id
+		}
 	}
 
 	onComplete() {
-		const holdDuration = this.uses === 0 ? this.parser.currentDuration : this.totalHeld
-		const usesMissed = Math.floor(holdDuration / this.data.actions.CELESTIAL_OPPOSITION.cooldown)
-		const fightDuration = this.parser.fight.end_time - this.parser.fight.start_time
-		const maxUses = (fightDuration / this.data.actions.CELESTIAL_OPPOSITION.cooldown) - 1
+		const holdDuration = this.uses === 0 ? this.parser.pull.duration : this.totalHeld
+		const missedUses = Math.floor(holdDuration / (this.data.actions.CELESTIAL_OPPOSITION.cooldown))
+		// TODO: update max uses to not use the whole duration such as when there are no available targets
+		const maxUses = (this.parser.pull.duration / this.data.actions.CELESTIAL_OPPOSITION.cooldown) - 1
 
 		const WASTED_USE_TIERS = {
-			[maxUses * SEVERITY_MOD.MINOR]: SEVERITY.MINOR,
-			[maxUses * SEVERITY_MOD.MEDIUM]: SEVERITY.MEDIUM,
-			[maxUses * SEVERITY_MOD.MAJOR]: SEVERITY.MAJOR, // if not used at all, it'll be set to 100 for severity checking
+			[maxUses * severityMod.minor]: SEVERITY.MINOR,
+			[maxUses * severityMod.medium]: SEVERITY.MEDIUM,
+			[maxUses * severityMod.major]: SEVERITY.MAJOR, // if not used at all, it'll be set to 100 for severity checking
```

Comment:
I feel like these should just be explicit, but it was also fine as it was before this change.

Reply:
reverted